File: lib/requests.rs
use serde::Serialize;
use crate::shared;

#[derive(Serialize, Debug)]
pub struct PurchaseOrderRequest {
    #[serde(rename(serialize = "shipId"))]
    pub ship_id: String,
    pub good: shared::Good,
    pub quantity: u32,
}

#[derive(Serialize, Debug)]
pub struct PurchaseShipRequest {
    pub location: String,
    #[serde(rename(serialize = "type"))]
    pub ship_type: String,
}

#[derive(Serialize, Debug)]
pub struct RequestNewLoanRequest {
    #[serde(rename(serialize = "type"))]
    pub loan_type: shared::LoanType,
}

#[derive(Serialize, Debug)]
pub struct FlightPlanRequest {
    #[serde(rename(serialize = "shipId"))]
    pub ship_id: String,
    pub destination: String,
}

File: spacetraders/requests.go
package spacetraders

type CreateFlightPlanRequest struct {
	ShipId      string `json:"shipId"`
	Destination string `json:"destination"`
}

type CreateLoanRequest struct {
	Type string `json:"type"`
}

type CreatePurchaseOrderRequest struct {
	ShipId   string `json:"shipId"`
	Good     string `json:"good"`
	Quantity int    `json:"quantity"`
}

type CreateSellOrderRequest struct {
	ShipId   string `json:"shipId"`
	Good     string `json:"good"`
	Quantity int    `json:"quantity"`
}

type PurchaseShipRequest struct {
	Location string `json:"location"`
	Type     string `json:"type"`
}

type JettisonCargoRequest struct {
	Good     string `json:"good"`
	Quantity int    `json:"quantity"`
}

type WarpJumpRequest struct {
	ShipId string `json:"shipId"`
}
